cluster-autoscaler/utils/deletetaint/delete.go

```diff
@@ -89,7 +91,20 @@ func addTaint(node *apiv1.Node, client kube_client.Interface, taintKey string, e
 			}
 			return nil
 		}
-		_, err = client.CoreV1().Nodes().Update(context.TODO(), freshNode, metav1.UpdateOptions{})
+
+		// update all taints, so we do not have to deal with adding to possibly nil taints or taint already existing
+		// patching also avoids overwriting changes that other cluster members might have done in the meantime
+		taintsJson, err := json.Marshal(freshNode.Spec.Taints)
+		if err != nil {
+			klog.Warningf("Error while adding %v taint on node %v: %v", getKeyShortName(taintKey), node.Name, err)
+			return err
+		}
+		patch := []byte(
+			`[{"op":"replace","path":"/spec/taints","value":` + string(taintsJson) + `},` +
+				` {"op":"replace","path":"/spec/unschedulable","value":` + strconv.FormatBool(freshNode.Spec.Unschedulable) + `}]`,
+		)
+
+		_, err = client.CoreV1().Nodes().Patch(context.TODO(), freshNode.Name, types.JSONPatchType, patch, metav1.PatchOptions{})
```

Comment:
Should we replace all `client.CoreV1().Nodes().Update()` to `client.CoreV1().Nodes().Patch()` or just here?

Reply:
ideally all ... but I'm trying to keep this PR small to get consensus on if this approach is good
and see if there is maybe a reason I'm overlooking for using `Update`
so ideally merge this once everyone agrees that it's good and then rework the others in the same way